src/main/java/org/sobotics/guttenberg/clients/Guttenberg.java

```diff
@@ -182,4 +198,49 @@ private void checkLastExecution() {
 		}
 		
 	}
+	
+	/**
+	 * Checks for updates
+	 * */
+	private void update() {
+		System.out.println("Load updater...");
+		Updater updater = new Updater();
+		System.out.println("Check for updates...");
+		
+		int update = updater.updateIfAvailable(); 
+		
+		if (update == -1) {
+			for (Room room : this.chatRooms) {
+				if (room.getRoomId() == 111347) {
+					room.send("Automatic update failed!");
+				}
+			}
+		} else if (update == 1) {
+			for (Room room : this.chatRooms) {
+				if (room.getRoomId() == 111347) {
+					room.send("Rebooting for update to version "+updater.getNewVersion().get());
+				}
+				room.leave();
+			}
+			System.exit(0);
+		}
+		
+	}
+	
+	
+	
+	private void setLogfile() {
+		Date now = Date.from(Instant.now());
+		String dateString = new SimpleDateFormat("yyyy-MM-dd-HH").format(now);
```

Comment:
With Java 8, you should drop the legacy `SimpleDateFormat` and only use `DateTimeFormatter`:

```java
String dateString = LocalDateTime.now(ZoneOffset.UTC).format(DateTimeFormatter.ofPattern("yyyy-MM-dd-HH"));
```